src/daemon/auth/providers/openid_connect/provider.rs

```diff
@@ -357,49 +363,84 @@ impl OpenIDConnectAuthProvider {
 
                                         match new_token_res {
                                             Ok(new_token) => {
-                                                return Some(Auth::Bearer(new_token));
+                                                // The new token was successfully acquired from the OpenID Connect Provider,
+                                                // and early returned.
+                                                return Ok(Auth::Bearer(new_token));
                                             }
                                             Err(err) => {
-                                                warn!(
-                                                    "OpenID Connect: Could not extend login session for user '{}': {}",
-                                                    &session.id, err
-                                                );
+                                                return Err(CoreErrorResponseType::Extension(format!(
+                                                    "Error while encoding the refreshed token {}",
+                                                    err
+                                                )));
                                             }
                                         }
                                     }
                                     Err(err) => {
```

Comment:
I think the code here is the same for any request to the OpenID Connect provider as I suspect they all return error responses of the same structure and with the same error categories. E.g. whether it be the call to exchange the temporary token for an access and id token, or the call to refresh a token, or the call to revoke a token (see PR #410). As such I wonder if this logic shouldn't be moved into something like `logging_http_client` in `util.rs` which is used for all calls to the provider made via the OpenID Connect crate?

Reply:
Maybe, yes, but not sure. The way I read it is that the Errors in Section 4.2.2 are only valid as responses from the redirect URL from the initial Authorization Request (4.2.1). Subsequential access token requests can have Error responses from Section 5.2. These two error sections are subtly different.

Reply:
Discussed with @density215. This might be the way to go but is a wider change that can be investigated as a separate PR.